File: GenesisAndroid/src/main/java/com/emerchantpay/gateway/genesisandroid/api/interfaces/financial/MpiAttributes.java
package com.emerchantpay.gateway.genesisandroid.api.interfaces.financial;

import com.emerchantpay.gateway.genesisandroid.api.util.RequestBuilder;

public interface MpiAttributes {

    RequestBuilder requestBuilder = new RequestBuilder("");

    // Mpi Params
    default MpiAttributes setCavv(String cavv) {
        requestBuilder.addElement("cavv", cavv);
        return this;
    }

    default MpiAttributes setEci(String eci) {
        requestBuilder.addElement("eci", eci);
        return this;
    }

    default MpiAttributes setXid(String xid) {
        requestBuilder.addElement("xid", xid);
        return this;
    }

    default RequestBuilder buildMpiParams() {
        return requestBuilder;
    }
}

File: GenesisAndroid/src/main/java/com/emerchantpay/gateway/genesisandroid/api/interfaces/financial/MpiAttributes.kt
package com.emerchantpay.gateway.genesisandroid.api.interfaces.financial

import com.emerchantpay.gateway.genesisandroid.api.util.RequestBuilder

interface MpiAttributes {

    // Mpi Params
    fun setCavv(cavv: String): MpiAttributes {
        requestBuilder.addElement("cavv", cavv)
        return this
    }

    fun setEci(eci: String): MpiAttributes {
        requestBuilder.addElement("eci", eci)
        return this
    }

    fun setXid(xid: String): MpiAttributes {
        requestBuilder.addElement("xid", xid)
        return this
    }

    fun buildMpiParams(): RequestBuilder {
        return requestBuilder
    }

    companion object {

        val requestBuilder = RequestBuilder("")
    }
}